include/cudaq/Optimizer/Dialect/Quake/QuakeOps.td

```diff
@@ -730,34 +844,36 @@ def HOp : OneTargetOp<"h", [Hermitian]> {
     //===------------------------------------------------------------------===//
     using Matrix = mlir::SmallVectorImpl<std::complex<double>>;
 
-    /// Set `matrix` to the operator's unitary matrix as a column-major array.
+    /// Set `matrix` to the operator's unitary matrix as a column-major array if
+    /// it can be computed at compilation time, i.e., if the parameter is a
+    /// compile-time constant.
     void getOperatorMatrix(Matrix &matrix) {
-      using namespace llvm::numbers;
-      matrix.assign({inv_sqrt2, inv_sqrt2, inv_sqrt2, -inv_sqrt2});
-    }
-
-    bool isClifford() {
-      return getControls().size() == 0;
+      using namespace std::complex_literals;
+      double theta;
+      if (failed(getParameterAsDouble(theta)))
+        return;
+      if (getIsAdj())
+        theta *= -1;
+      matrix.assign({      std::cos(theta / 2.), -1i * std::sin(theta / 2.),
+                     -1i * std::sin(theta / 2.), std::cos(theta / 2.)});
```

Comment:
I think we should move C++ code like this to the QuakeOps.cpp file.

Reply:
We can. The current implementation is like to facilitate sharing between Quake and QTX.